File: 0999-available-captures-for-rook/0999-available-captures-for-rook.ts
function numRookCaptures(board: string[][]): number {
    let rr: number | null
    let rc: number | null
    let res = 0
    for (let r=0; r < board.length; r++) {
        for (let c=0; c < board[r].length; c++) {
            if (board[r][c] === "R") {
                rr = r
                rc = c
                break
            }
        }
    }
    for (const [r, c] of [[0, 1], [1, 0], [0, -1], [-1, 0]]) {
        for (let i=0; i < board.length; i++) {
            const rrr = (r * i) + rr
            const ccc = (c * i) + rc
            if (!(rrr in board) || !(ccc in board) || board[rrr][ccc] === "B") {
                break
            }
            if (board[rrr][ccc] === "p") {
                res++
                break
            }
        }
    }
    return res
};

File: 0999-available-captures-for-rook/0999-available-captures-for-rook.py
class Solution:
    def numRookCaptures(self, board: List[List[str]]) -> int:
        pr, pc, res = None, None, 0
        for r in range(0, len(board)):
            for c in range(0, len(board[r])):
                if board[r][c] == "R":
                    pr = r
                    pc = c
                    break
        for r, c in [[1, 0], [0, 1], [-1, 0], [0, -1]]:
            for i in range(0, len(board)):
                rr = pr + (r * i)
                cc = pc + (c * i)
                if rr < 0 or cc < 0 or rr >= len(board) or cc >= len(board[rr]) or board[rr][cc] == "B":
                    break
                if board[rr][cc] == "p":
                    res += 1
                    break
        return res